packages/cli/src/deploy/getResourceIds.ts

```diff
@@ -15,15 +17,25 @@ export async function getResourceIds({
   // TODO: PR to viem's getLogs to accept topics array so we can filter on all store events and quickly recreate this table's current state
 
   debug("looking up resource IDs for", worldDeploy.address);
-  const logs = await getLogs(client, {
-    strict: true,
-    address: worldDeploy.address,
-    fromBlock: worldDeploy.deployBlock,
-    toBlock: worldDeploy.stateBlock,
-    event: parseAbiItem(storeSpliceStaticDataEvent),
-    args: { tableId: storeTables.store_ResourceIds.tableId },
-  });
-
+  const logs = await pRetry(
+    () =>
+      getLogs(client, {
+        strict: true,
+        address: worldDeploy.address,
+        fromBlock: worldDeploy.deployBlock,
+        toBlock: worldDeploy.stateBlock,
+        event: parseAbiItem(storeSpliceStaticDataEvent),
+        args: { tableId: storeTables.store_ResourceIds.tableId },
+      }),
+    {
+      retries: 3,
+      onFailedAttempt: async (error) => {
```

Comment:
we could check the error to see if it contains "block is out of range" and only retry those? but I am happy just retrying all here since its a relatively safe operation

Reply:
Done, I also like that it makes the retry logic more self-explanatory